Origin: http://websvn.kde.org/?view=revision&revision=1227468
Description: Further addresses CVE-2010-1000.
 The file name of Metalink File is checked a better way, making it work under
 more conditions.
--- a/kget/ui/metalinkcreator/metalinker.cpp
+++ b/kget/ui/metalinkcreator/metalinker.cpp
@@ -583,7 +583,13 @@
         return false;
     }
 
-    if (name.contains(QRegExp("$(\\.\\.?)?/")) || name.contains("/../") || name.endsWith("/..")) {
+    if (name.endsWith('/')) {
+        kError(5001) << "Name attribute of Metalink::File does not contain a file name:" << name;
+        return false;
+    }
+
+    const QStringList components = name.split('/');
+    if (name.startsWith('/') || components.contains("..") || components.contains(".")) {
         kError(5001) << "Name attribute of Metalink::File contains directory traversal directives:" << name;
         return false;
     }
